TopQuarkAnalysis/TopEventProducers/src/PseudoTopProducer.cc

```diff
@@ -54,15 +54,15 @@ void PseudoTopProducer::produce(edm::Event& event, const edm::EventSetup& eventS
   auto pseudoTopRefHandle = event.getRefBeforePut<reco::GenParticleCollection>();
 
   // Collect unstable B-hadrons
-  std::set<size_t> bHadronIdxs;
+  std::set<int> bHadronIdxs;
   for ( size_t i=0, n=genParticleHandle->size(); i<n; ++i )
   {
     const reco::Candidate& p = genParticleHandle->at(i);
     const int status = p.status();
     if ( status == 1 ) continue;
 
     // Collect B-hadrons, to be used in b tagging
-    if ( isBHadron(&p) ) bHadronIdxs.insert(i);
+    if ( isBHadron(&p) ) bHadronIdxs.insert(-i);
```

Comment:
It can be a matter of convention, but how about store the index as is `bHadronIdxs.insert(i);` here and `set_user_index(-index-1)`?
- subtracting by another 1 to make the user index to be negative-definite to be 100% safe from overlapping.